### PR TITLE
Add variance thresholding

### PR DESCRIPTION
**Pull request by jackyho112**
I banged this out in a short amount of time, so I might miss some edge cases or my logic might be flawed in some ways. Please look it over and see if anything is amiss. Also, to test, run `pytest` in your console in the project folder. Thanks!!!

### CODE REVIEW

File: tests/test_variance_threshold_select.py [hunk 1]
```diff
@@ -0,0 +1,38 @@
+from feature_selection import variance_threshold_select
+import numpy as np
+import seaborn as sns
+import pandas as pd
+import pytest
+
+iris = sns.load_dataset('iris')
```
Comment (techrah) [inline tests/test_variance_threshold_select.py]: Can we load the iris dataset from [sklearn](https://scikit-learn.org/stable/modules/generated/sklearn.datasets.load_iris.html) instead? This way we don't need to include seaborn.

File: feature_selection/variance_threshold_select.py [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    >>> variance_threshold_select(X)
+    array([1, 2])
+    """
+
+    is_data_list = isinstance(data, list)
+    is_data_df = isinstance(data, pd.DataFrame)
+    is_data_np_array = isinstance(data, np.ndarray)
+
+    if not (is_data_list or is_data_df or is_data_np_array):
+        raise Exception('Data is of an invalid type.')
```
Comment (techrah) [inline feature_selection/variance_threshold_select.py]: Let's use `TypeError` for consistency with scikit-learn

```suggestion
        raise TypeError('Data is of an invalid type.')
```

File: feature_selection/variance_threshold_select.py [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+    elif (is_data_np_array or is_data_list) and len(np.array(data).shape) > 2:
+        raise Exception(
+            'Data is of an invalid shape. ' \
+            'Please only pass in data of less than two dimensions.'
+        )
```
Comment (techrah) [inline feature_selection/variance_threshold_select.py]: Let's use `ValueError` for consistency with scikit-learn

```suggestion
        raise ValueError(
            'Data is of an invalid shape. ' \
            'Please only pass in data of less than two dimensions.'
        )
```

**Review by techrah — CHANGES_REQUESTED** [review]
Minor requests.

We also need to add package dependencies. However, my pull request #31 already includes pandas and numpy so no need to do it here.

**Review by techrah — APPROVED** [review]

**Review by techrah — APPROVED** [review]
Looks good!
